src/main/java/org/folio/oaipmh/WebClientProvider.java

```diff
@@ -0,0 +1,82 @@
+package org.folio.oaipmh;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.folio.oaipmh.Constants.REPOSITORY_SRS_CLIENT_IDLE_TIMEOUT_SEC;
+
+public class WebClientProvider {
+
+  private static final Logger logger = LogManager.getLogger(WebClientProvider.class);
+
+  private static final int DEFAULT_IDLE_TIMEOUT_SEC = 20;
+  private static final int DEFAULT_CONNECTION_TIMEOUT_MS = 2000;
+  private static final int DOWNLOAD_INSTANCE_CONNECTION_TIMEOUT_MS = 2400000;
+  private static final int KEEP_ALIVE_SEC = 1200;
+  private static final String GET_IDLE_TIMEOUT_ERROR_MESSAGE = "Error occurred during resolving the idle timeout setting value. Setup client with default idle timeout " + DEFAULT_IDLE_TIMEOUT_SEC + " seconds.";
+
+  private static Vertx vertx;
+  private static WebClient webClient;
+  private static WebClient webClientToDownloadInstances;
+  private static final Map<String, WebClient> webClientForSRSPerTenant = new ConcurrentHashMap<>();
+
+  private WebClientProvider() {}
+
+  public static void init(Vertx v) {
+    vertx = v;
+    webClient = WebClient.create(vertx);
+    WebClientOptions options = new WebClientOptions()
+      .setKeepAliveTimeout(KEEP_ALIVE_SEC)
+      .setConnectTimeout(DOWNLOAD_INSTANCE_CONNECTION_TIMEOUT_MS);
+    webClientToDownloadInstances = WebClient.create(vertx, options);
+  }
+
+  public static WebClient getWebClient() {
+    return webClient;
+  }
+
+  public static WebClient getWebClientToDownloadInstances() {
+    return webClientToDownloadInstances;
+  }
+
+  public static WebClient getWebClientForSRSByTenant(String tenant) {
+    return webClientForSRSPerTenant.computeIfAbsent(tenant, WebClientProvider::createWebClientWithSRSConfiguredOptions);
+  }
+
+  public static void closeAll() {
+    webClient.close();
+    webClientToDownloadInstances.close();
+    webClientForSRSPerTenant.values().forEach(WebClient::close);
+  }
+
+  private static WebClient createWebClientWithSRSConfiguredOptions(String tenant) {
+    String property = System.getProperty(REPOSITORY_SRS_CLIENT_IDLE_TIMEOUT_SEC);
+    final String defaultValue = Objects.nonNull(property) ? property : String.valueOf(DEFAULT_IDLE_TIMEOUT_SEC);
+    String val = Optional.ofNullable(vertx.getOrCreateContext()
+      .config()
+      .getJsonObject(tenant))
+      .map(config -> config.getString(REPOSITORY_SRS_CLIENT_IDLE_TIMEOUT_SEC, defaultValue))
+      .orElse(defaultValue);
```

Comment:
It is required to take a config for SRS client setup from RepositoryConfigurationUtil configMap instead of the default one.
```suggestion
    String val = RepositoryConfigurationUtil.getProperty(requestId, REPOSITORY_SRS_CLIENT_IDLE_TIMEOUT_SEC);
```